File: args-python/application/exceptions.py
class ParserException(Exception):
    pass


class UnknownArgumentException(ParserException):
    def __init__(self, argument: str):
        self.argument = argument

    def __str__(self):
        return f"Unknown argument provided: {self.argument}"

class InvalidArgumentValueException(ParserException):
    def __init__(self, argument, value):
        self.argument = argument
        self.value = value

    def __str__(self):
        return f"Invalid argument value provided: {self.argument} {self.value}"

File: args-kotlin/src/main/kotlin/nl/mvdb/args/Exceptions.kt
package nl.mvdb.args

import java.lang.RuntimeException

open class ParserException(message: String) : RuntimeException(message)

class UnknownIdentifierException(identifier: String) : ParserException("Unknown argument provided: $identifier")

class InvalidArgumentValueException(identifier: String, value: String?) : ParserException("Invalid argument value provided: $identifier $value")
